### PR TITLE
CFE-4063: Added link targets for variable class expressions and automatic canonification

### PR DESCRIPTION
**Pull request by nickanderson**
Ticket: CFE-4063
Changelog: Title

### CODE REVIEW

File: reference/language-concepts/classes.markdown [hunk 1]
```diff
@@ -343,6 +354,7 @@ bundle agent greetings
   reports:
    "$(myclassname)"::
      "Good evening!";
+     "What a wonderful sunset";
```
Comment (craigcomstock) [inline reference/language-concepts/classes.markdown:357]: Why is this added here? I don't see outputs from this bundle or an explanation of what this second reports promise "shows" about how variables used in quoted class expressions works. :)

Comment (nickanderson) [reply reference/language-concepts/classes.markdown:357]: Ah, ill update the output. Good. catch. UGH I HATE INLINE examples.

But, the addition was to speak to this: https://github.com/cfengine/documentation/pull/2906/files#diff-596050329d2a43aa188f8c3d1b06d7aa6d014ca65e3f7dde480e8f992dc807b3R338

> Variables used in quoted class expressions apply to all promises following the
quoted class expression until the end of the promise type or until a new class
expression is stated.
